lib/gym/generators/build_command_generator.rb

```diff
@@ -90,16 +92,19 @@ def build_path
       end
 
       def archive_path
-        Gym.cache[:archive_path] ||= Gym.config[:archive_path]
-        unless Gym.cache[:archive_path]
-          file_name = [Gym.config[:output_name], Time.now.strftime("%F %H.%M.%S")] # e.g. 2015-08-07 14.49.12
-          Gym.cache[:archive_path] = File.join(build_path, file_name.join(" ") + ".xcarchive")
-        end
-
-        if File.extname(Gym.cache[:archive_path]) != ".xcarchive"
-          Gym.cache[:archive_path] += ".xcarchive"
+        if Gym.config[:archive]
+          Gym.cache[:archive_path] ||= Gym.config[:archive_path]
+          unless Gym.cache[:archive_path]
+            file_name = [Gym.config[:output_name], Time.now.strftime("%F %H.%M.%S")] # e.g. 2015-08-07 14.49.12
+            Gym.cache[:archive_path] = File.join(build_path, file_name.join(" ") + ".xcarchive")
+          end
+
+          if File.extname(Gym.cache[:archive_path]) != ".xcarchive"
+            Gym.cache[:archive_path] += ".xcarchive"
+          end
+          return Gym.cache[:archive_path]
         end
-        return Gym.cache[:archive_path]
+        return ""
```

Comment:
Not sure we want this feature, but if we want it, shouldn t it be more logical to add
```
raise "No archive path available when " unless Gym.config[:archive] 
```

at the top of this method ?